openstates/ut/legislators.py

```diff
@@ -1,125 +1,81 @@
 from billy.scrape.legislators import LegislatorScraper, Legislator
-
+from scrapelib import HTTPError
 import lxml.html
 
+from openstates.utils import LXMLMixin
+
 
-class UTLegislatorScraper(LegislatorScraper):
+class UTLegislatorScraper(LegislatorScraper,LXMLMixin):
     jurisdiction = 'ut'
     latest_only = True
 
-    def scrape(self, chamber, term):
-        if chamber == 'lower':
-            self.scrape_lower(term)
-        else:
-            self.scrape_upper(term)
-
-
-    def scrape_lower(self, term):
-        url = 'http://le.utah.gov/house2/representatives.jsp'
-        html = self.get(url).text
-        doc = lxml.html.fromstring(html)
-        doc.make_links_absolute(url)
-
-        for row in doc.xpath('//tr')[1:]:
-            tds = row.xpath('td')
-
-            district = tds[0].text_content()
-            if tds[1].text_content() == 'Empty':
-                self.log('district %s is empty' % district)
-                continue
-            a = tds[1].xpath('a')[0]
-            name = a.text_content()
-            leg_url = a.get('href')
-
-            party = tds[2].text_content()
-            if party == 'D':
-                party = 'Democratic'
-            elif party == 'R':
-                party = 'Republican'
-            else:
-                raise ValueError('unknown party')
-
-            # get photo
-            leg_html = self.get(leg_url).text
-            leg_doc = lxml.html.fromstring(leg_html)
-            leg_doc.make_links_absolute(leg_url)
-            photo_url = leg_doc.xpath('//img[@alt="photo"]/@src')[0]
-            email = leg_doc.xpath('//a[starts-with(@href, "mailto")]')[0].text
-
-            address = leg_doc.xpath('//b[text()="Address:"]')[0].tail.strip()
-            cell = leg_doc.xpath('//b[text()="Cell Phone:"]')
-            if cell:
-                cell = cell[0].tail.strip()
+    def scrape(self, term, chambers):
+        #utah seems to have undocumented JSON!
+        house_base_url = "http://le.utah.gov/house2/"
+        senate_base_url = "http://senate.utah.gov/"
+        json_link = "http://le.utah.gov/data/legislators.json"
+        leg_json = self.get(json_link).json()
+        for leg_info in leg_json["legislators"]:
+            leg_name = leg_info["fullName"]
```

Comment:
Using `formatName` instead of `fullName` seems a tiny bit better, since it reports legislators' names in a clean {first} {middle} {last} order, instead of {last}, {first} {middle}.